.azure-pipelines/ci-build.yml

```diff
@@ -12,235 +12,185 @@ pr:
-
-      # CredScan
-      - task: securedevelopmentteam.vss-secure-development-tools.build-task-credscan.CredScan@3
-        displayName: 'Run CredScan - Src'
-        inputs:
-          toolMajorVersion: 'V2'
-          scanFolder: '$(Build.SourcesDirectory)\src'
-          debugMode: false
-
-      - task: securedevelopmentteam.vss-secure-development-tools.build-task-credscan.CredScan@3
-        displayName: 'Run CredScan - Test'
-        inputs:
-          toolMajorVersion: 'V2'
-          scanFolder: '$(Build.SourcesDirectory)\test'
-          debugMode: false
-
-      - task: AntiMalware@3
-        displayName: 'Run MpCmdRun.exe - ProductBinPath'
-        inputs:
-          FileDirPath: '$(ProductBinPath)'
-        enabled: false
-
-      - task: BinSkim@4
-        displayName: 'Run BinSkim - Product Binaries'
-        inputs:
-          InputType: Basic
-          AnalyzeTargetGlob: '$(ProductBinPath)\**\Microsoft.OpenApi.OData.Reader.dll'
-          AnalyzeSymPath: '$(ProductBinPath)'
-          AnalyzeVerbose: true
-          AnalyzeHashes: true
-          AnalyzeEnvironment: true
-
-      - task: PublishSecurityAnalysisLogs@3
-        displayName: 'Publish Security Analysis Logs'
-        inputs:
-          ArtifactName: SecurityLogs
-
-      - task: PostAnalysis@2
-        displayName: 'Post Analysis'
-        inputs:
-          BinSkim: true
-          CredScan: true
-          PoliCheck: true
-
-      - task: SFP.build-tasks.custom-build-task-1.EsrpCodeSigning@1
-        displayName: 'ESRP CodeSigning'
-        inputs:
-          ConnectedServiceName: 'microsoftgraph ESRP CodeSign DLL and NuGet (AKV)'
-          FolderPath: src
-          signConfigType: inlineSignParams
-          inlineOperation: |
-            [
-                {
-                    "keyCode": "CP-230012",
-                    "operationSetCode": "SigntoolSign",
-                    "parameters": [
-                    {
-                        "parameterName": "OpusName",
-                        "parameterValue": "Microsoft"
-                    },
+resources:
+  repositories:
+  - repository: 1ESPipelineTemplates
+    type: git
+    name: 1ESPipelineTemplates/1ESPipelineTemplates
+    ref: refs/tags/release
+
+extends:
+  template: v1/1ES.Official.PipelineTemplate.yml@1ESPipelineTemplates
+  parameters:
+    pool:
+      name: Azure-Pipelines-1ESPT-ExDShared
+      image: windows-2022
+      os: windows
+      demands:
+      - msbuild
+      - vstest
+    customBuildTags:
+    - ES365AIMigrationTooling 
+    stages:
+
+    - stage: build
+      jobs:
+        - job: build
+          templateContext:
+            outputs:
+            - output: pipelineArtifact
+              displayName: 'Publish Artifact: Nugets'
+              artifactName: Nugets
+              targetPath: '$(Build.ArtifactStagingDirectory)'
+          steps:
+
+          - task: UseDotNet@2
+            displayName: 'Use .NET 2' # needed for ESRP signing
```

Comment:
We should probably change this to net 6 then update the signing tasks to the latest `EsrpCodeSigning@3`